docs/2022-isc22/troubleshooting.md

```diff
@@ -1 +1,485 @@
+
+The location of the build directory is mentioned in the EasyBuild error message:
+
+```
+== FAILED: Installation ended unsuccessfully (build directory: /tmp/example/example/1.0/GCC-10.2.0): ...
+```
+
+For software using a classic `configure` script, you may have to locate
+and inspect the `config.log` file in the build directory to determine the underlying cause of an error. For software using CMake as a configuration tool you often have to check in
+`CMakeOutput.log` or `CMakeError.log` for clues, which are sneakily hidden by CMake in a `CMakeFiles` subdirectory of the build directory.
+
+As a side note here: as EasyBuild does not clean out build directories for failed builds, you will need to eventually manually remove them from the `buildpath` directory.
+
+## Exercise
+
+Let's work our way through a less smooth software installation,
+using the easyconfig file that is provided below.
+
+Don't worry if most of this is still unclear to you, we'll get
+to writing easyconfig files from scratch [later in this tutorial](adding_support_additional_software.md).
+
+```python
+easyblock = 'MakeCp'
+
+name = 'Subread'
+version = '2.0.1'
+
+homepage = 'http://subread.sourceforge.net'
+description = "High performance read alignment, quantification and mutation discovery"
+
+toolchain = {'name': 'GCC', 'version': '8.5.0'}
+
+# download from https://download.sourceforge.net/subread/subread-2.0.1-source.tar.gz
+sources = ['subread-%(version)s-source.tar.gz']
+checksums = ['d808eb5b1823c572cb45a97c95a3c5acb3d8e29aa47ec74e3ca1eb345787c17b']
+
+start_dir = 'src'
+
+# -fcommon is required to compile Subread 2.0.1 with GCC 10,
+# which uses -fno-common by default (see https://www.gnu.org/software/gcc/gcc-10/porting_to.html)
+buildopts = '-f Makefile.Linux CFLAGS="-fast -fcommon"'
+
+files_to_copy = ['bin']
+
+sanity_check_paths = {
+    'files': ['bin/featureCounts', 'bin/subread-align'],
+    'dirs': ['bin/utilities'],
+}
+
+sanity_check_commands = ["featureCounts --version"]
+
+moduleclass = 'bio'
+```
+
+Do you spot any potential problems yet with this easyconfig file?
+
+***Preparation***
+
+Start by copying the text above in a file named `subread.eb`,
+so you can gradually fix the problem you'll encounter.
+
+Also make sure that the pre-installed software stack is available,
+and that the EasyBuild module is loaded (unless you installed EasyBuild
+yourself):
+
+```
+module use /easybuild/modules/all
+module load EasyBuild
+```
+
+For this exercise, make sure EasyBuild is configured to
+use `$HOME/easybuild` as `prefix`, and to use `/tmp/$USER` as `buildpath`:
+
+```shell
+export EASYBUILD_PREFIX=$HOME/easybuild
+export EASYBUILD_BUILDPATH=/tmp/$USER
+```
+
+Check your configuration via `eb --show-config`.
+
+Strictly speaking the configuration doesn't matter much for the sake of this
+exercise, but it may help with the step-wise approach we'll take and
+grasping the solutions.
```

Comment:
I wonder if we should add a warning that depending on how EasyBuild is used, a user may actually bump into T.2 before T.1? If you have the automatism to first check for dependencies, e.g., with eb -D, you'll actually first run in T.2, but if you just do `eb subread.eb` you'll run in T.1 first. Just a thought though, not a necessity.